virttest/qemu_devices/qdevice_format.py

```diff
@@ -0,0 +1,554 @@
+                },
+                "pvpanic": {
+                    "events": self._str_to_dec,
+                },
+            },
+            "object": {
+                "size": self._normalize_data_size,
+                "align": self._normalize_data_size,
+                "host-nodes": self._int_in_list,
+                "prealloc-threads": self._str_to_dec,
+            },
+            "netdev": {
+                "fd": self._unchanged,
+                "vhostfd": self._unchanged,
+                "dnssearch": self._dict_in_list,
+                "hostfwd": self._dict_in_list,
+                "guestfwd": self._dict_in_list,
+                "sndbuf": self._normalize_data_size,
+            },
+        }
+        self._device_driver_checked = []
+        # _skip_args stores those arguments which are NOT expected to be
+        # converted while updating the arguments type from qemu on machine.
+        #   About "addr" in qemu output,
+        #       addr=<int32> - Slot and optional function number, example: 06.0 or 06 (default: -1)
+        #   In fact: The "addr" is accepted by hexadecimal in string instead of
+        #   hexadecimal in int.
+        #   The following is qemu output:
+        #   qemu-kvm: -device {"id": "pcie-root-port-0", "driver": "pcie-root-port",
+        #   "multifunction": true, "bus": "pcie.0", "addr": 1,
+        #   "chassis": 1}: PCI: single function device can't be populated in function 0.1
+        self._skip_args = ("addr",)
+        # _mandatory_assignment_args_type stores those arguments whose types
+        # are assigned. The _mandatory_assignment_args_type structure is as
+        # same as the _special_args_in_json structure.
+        self._mandatory_assignment_args_type = {
+            "device": {
+                "pvpanic": {
+                    #   About "ioport" in qemu output:
+                    #       ioport=<uint16>        -  (default: 1285)
+                    #   In fact: The "ioport" is accepted by function
+                    #   _hex_in_str_to_dec instead of function _str_to_dec.
+                    "ioport": self._hex_in_str_to_dec,
+                },
+                "vhost-vsock-pci": {
+                    #   About "guest-cid" in qemu output:
+                    #       guest-cid=<uint64>     -  (default: 0)
+                    #   In fact: The "guest-cid" is accepted by function
+                    #   _str_to_dec instead of function _hex_in_str_to_dec.
+                    "guest-cid": self._str_to_dec,
+                },
+            }
+        }
+        self._type_func_mapping = {
+            "<int16>": self._str_to_dec,
+            "<uint16>": self._str_to_dec,
+            "<int32>": self._str_to_dec,
+            "<uint32>": self._str_to_dec,
+            "<bool>": self._bool_in_string_to_bool,
+            "<int64>": self._hex_in_str_to_dec,
+            "<uint64>": self._hex_in_str_to_dec,
```

Comment:
@nickzhq 
I have some question about how we handle uint64 here:
/usr/libexec/qemu-kvm -device vhost-vsock-ccw,?
vhost-vsock-ccw options:
  any_layout=<bool>      - on/off (default: true)
  dev_id=<str>           - Read-only identifier of an I/O device in the channel subsystem, example: fe.1.23ab
  devno=<str>            - Identifier of an I/O device in the channel subsystem, example: fe.1.23ab
  event_idx=<bool>       - on/off (default: true)
  guest-cid=<uint64>     -  (default: 0)
and how we deal with it here will return (val, 16) to fail the test case
I'll add the test result in jira issue

Reply:
> @nickzhq I have some question about how we handle uint64 here: /usr/libexec/qemu-kvm -device vhost-vsock-ccw,? vhost-vsock-ccw options: any_layout= - on/off (default: true) dev_id= - Read-only identifier of an I/O device in the channel subsystem, example: fe.1.23ab devno= - Identifier of an I/O device in the channel subsystem, example: fe.1.23ab event_idx= - on/off (default: true) guest-cid= - (default: 0) and how we deal with it here will return (val, 16) to fail the test case I'll add the test result in jira issue

Thanks to @fbq815 
After discussed with @fbq815 , this patch will support args of `vhost-vsock-ccw` to test.
However, we may consider this is a bug for qemu. 
If we have any updates, will put at here.

Reply:
approved this was fixed:
qemu.vsock_test.hotplug.s390-virtio: PASS (136.63 s)